Raven.Client.Lightweight/Changes/RemoteChangesClientBase.cs

```diff
@@ -201,83 +213,133 @@ private bool DisposeHeartbeatTimer()
             }
         }
 
-        private Task lastSendTask;
-
-        protected Task Send(string command, string value)
+        async Task Worker()
         {
-            lock (this)
+            while (tokenSource.IsCancellationRequested == false)
             {
-                logger.Info("Sending command {0} - {1} to {2} with id {3}", command, value, url, id);
-                var sendTask = lastSendTask;
-                if (sendTask != null)
+                try
                 {
-                    return sendTask.ContinueWith(_ =>
-                    {
-                        Send(command, value);
-                    });
+                    await syncSource.Task.ConfigureAwait(false);
+                    syncSource = TaskCompletionSourceFactory.Create<bool>();
```

Comment:
You are trying to do a manual reset event, no?
Better to use:
https://github.com/ravendb/ravendb/blob/v3.5/Raven.Abstractions/Util/AsyncManualResetEvent.cs

Reply:
Doesn't seem to use the TCS factory and therefore doesn't use RunContinuationsAsync